build/build.yml

```diff
@@ -2,14 +2,18 @@ parameters:
   packageArtifacts: true
   
 steps:
+- template: ./update-semver.yml
 
 - task: UseDotNet@2
   displayName: 'Use .NET Core sdk'
   inputs:
     useGlobalJson: true
 
-- script: dotnet build --configuration $(buildConfiguration) --version-suffix $(build.buildNumber) /warnaserror
+- task: DotNetCoreCLI@2
   displayName: 'dotnet build $(buildConfiguration)'
+  inputs:
+    command: build
+    arguments: '--configuration $(buildConfiguration) /p:AssemblyVersion="$(assemblySemVer)" /p:FileVersion="$(assemblySemFileVer)" /p:InformationalVersion="$(informationalVersion)" /p:Version="$(majorMinorPatch)" /warnaserror'
```

Comment:
When we change the assembly version, the code will not be binary-compatible. A recompilation of the consuming app will be required. 
So we should maybe be clear that even in cases were we only increment the patch, we will not be binary-compatible.

Reply:
We could potentially trim this to have a fixed Minor.Patch, e.g. {Major}.0.0? Perhaps this makes more sense.
I think leaving the informational version as-is should be fine to inspect the origin

Reply:
The problem is that it's really easy to introduce changes that require recompilation but that are binary compatible. So I don't think we should do this. 
By the way, what is out level of commitment to not making breaking changes without updating the major number? Should we have a tool that checks the API surface?

Reply:
Makes sense, I'll revert the version string to a hard-coded "1.0.0" but keep the informational version.

We are also not publishing any packages to the official nuget feed yet, so I see some opportunity to keep improving here. I like the suggestion, do you know of any tools already? I'll can look into this

Reply:
With changes from above:
![image](https://user-images.githubusercontent.com/197221/100905085-819bea00-347c-11eb-9813-67de3df500f0.png)